pycbc/inference/sampler.py

```diff
@@ -26,6 +26,62 @@
 packages for parameter estimation.
 """
 
+import numpy
+from pycbc.io import WaveformArray
+
+#
+# =============================================================================
+#
+#                                   Helper functions
+#
+# =============================================================================
+#
+def get_slice(fp, thin_start=None, thin_interval=None, thin_end=None):
+    """Formats a slice using the given arguments that can be used to retrieve
+    a thinned array from an InferenceFile.
+
+    Parameters
+    ----------
+    fp : InferenceFile
+        An open inference file. The `burn_in_iterations` and acl will try to
+        be obtained from the file's attributes.
+    thin_start : {None, int}
+        The starting index to use. If None, will try to retrieve the
+        `burn_in_iterations` from the given file. If no `burn_in_iterations`            exists, will default to the start of the array.
+    thin_interval : {None, int}
+        The interval to use. If None, will try to retrieve the acl from the
+        given file. If no acl attribute exists, will default to 1.
+    thin_end : {None, int}
+        The end index to use. If None, will retrieve to the end of the array.
+
+    Returns
+    -------
+    slice :
+        The slice needed.
+    """
+    # default is to skip burn in samples
+    if thin_start is None:
+        try:
+            thin_start = fp.burn_in_iterations
+        except KeyError:
+            pass
+    # default is to use stored ACL and accept every i-th sample
+    if thin_interval is None:
+        try:
+            thin_interval = int(numpy.ceil(fp.acl))
+        except KeyError:
+            pass
+    return slice(thin_start, thin_end, thin_interval)
```

Comment:
I was doing some more tests since this is a large PR that touches a lot of important functions.

Currently plotting a running sampler is broken in some cases. Please include a block like:
```
# default is to use the most recent stored sample
if thin_end is None:
    try:
        thin_end = fp.niterations
    except KeyError:
        pass
```